### PR TITLE
#5255: fix var analysis false positives and enable var analysis

### LINKED ISSUE #5255: Fix VarAnalysis false positives
Areas to Fix/Add Test Coverage
---
- Triggers don't have accurate schema based on trigger type (event, etc.)
- Dynamic Quick Bar doesn't include "query"
- Element when inside a "list" in the document builder
- Double-check control flow bricks
  - Try-Except
  - For Each Element
  - For Each 

Related
---
- https://github.com/pixiebrix/pixiebrix-extension/issues/4723

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Closes #5255 
- Adds handling for Document Builder List Element to varAnalysis
- Adds reader metadata for Trigger and Dynamic Quick Bar starter bricks
- Modified `setInputVars` to use the defaultReader method from ExtensionPoint instance object
- Adds a `previewReader` method to IExtensionPoint for returning the reader to be user when generating Page Editor previews

## Included in PR

- [x] Remove skunkworks flag for var analysis
- [x] Add test for Try-Except brick
- [x] Add test for For-Each Element brick
- [x] Fix handling of Document Builder List Element
- [x] Fix Dynamic Quick Bar `@input.query`
- [x] Fix Trigger `@input.event`
- [x] ~~Fix Trigger `@input.element`~~ already working

## Remaining Work

- [x] Fix preview of Dynamic Quick Bar Inputs
- [x] Ensure preview of Trigger works

## Demo

- https://www.loom.com/share/0b1ae728b0c448d8a0ced2467b01e6eb

## Future Work

- In List Element, pass through the shape of the list/array element. Right now it just allows anything on the element
- In the For Each brick, pass through the shape of the array items if known. Right now it just allows any prop on the element

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer: @BLoe

### CODE REVIEW

File: src/extensionPoints/helpers.ts [hunk 1]
```diff
@@ -225,46 +221,6 @@ export function selectExtensionContext(
   };
 }
 
-export function pickEventProperties(nativeEvent: Event): JsonObject {
```
Comment (twschiller) [inline src/extensionPoints/helpers.ts:228]: Moved to triggerEventReaders file

File: src/extensionPoints/triggerExtension.ts [hunk 2]
```diff
@@ -62,97 +62,25 @@ import { PromiseCancelled } from "@/errors/genericErrors";
...
+  getEventReader,
+  pickEventProperties,
+} from "@/extensionPoints/triggerEventReaders";
+import CompositeReader from "@/blocks/readers/CompositeReader";
 
 export type TriggerConfig = {
   action: BlockPipeline | BlockConfig;
 };
 
-export type AttachMode =
```
Comment (twschiller) [inline src/extensionPoints/triggerExtension.ts:70]: Moved to triggerExtensionTypes

File: src/contentScript/pageEditor/dynamic.ts [hunk 3]
```diff
@@ -58,42 +53,6 @@ export async function clearDynamicElements({
   }
 }
 
-/**
```
Comment (twschiller) [inline src/contentScript/pageEditor/dynamic.ts:61]: Moved to contextMenuReader

File: src/core.ts [hunk 4]
```diff
@@ -705,8 +705,21 @@ export interface IExtensionPoint extends Metadata {
...
+   * @see IExtensionPoint.previewReader
+   */
   defaultReader: () => Promise<IReader>;
 
+  /**
+   * Return a IReader for generated an `@input` preview. The shape will correspond to defaultReader, but some
+   * properties may not be available
+   * @see defaultReader
+   */
+  previewReader: () => Promise<IReader>;
```
Comment (twschiller) [inline src/core.ts:721]: This is the new method exposed for generating a preview-compatible version of the reader

File: src/extensionPoints/triggerExtensionTypes.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+export type AttachMode =
```
Comment (twschiller) [inline src/extensionPoints/triggerExtensionTypes.ts:18]: Moved from triggerExtension.ts

File: src/options/pages/settings/ExperimentalSettings.tsx [hunk 6]
```diff
@@ -115,23 +114,9 @@ const ExperimentalSettings: React.FunctionComponent = () => {
               );
             }}
           />
-          <ExperimentalFeature
```
Comment (twschiller) [inline src/options/pages/settings/ExperimentalSettings.tsx:118]: Enabling for everyone

**Review by BLoe — APPROVED** [review]

**Pull request merged**